src/Microsoft.OpenApi/Extensions/OpenApiSerializableExtensions.cs

```diff
@@ -107,7 +107,7 @@ public static Task SerializeAsync<T>(
         /// <param name="writer">The output writer.</param>
         /// <param name="specVersion">Version of the specification the output should conform to</param>
         /// <param name="cancellationToken">The cancellation token.</param>
-        public static Task SerializeAsync<T>(this T element, IOpenApiWriter writer, OpenApiSpecVersion specVersion, CancellationToken cancellationToken = default)
+        public static Task SerializeAsync<T>(this T? element, IOpenApiWriter writer, OpenApiSpecVersion specVersion, CancellationToken cancellationToken = default)
```

Comment:
revert the change on element and the null propagation since we check that element is not null in the first line